packages/sol-doc/src/solidity_doc_generator.ts

```diff
@@ -0,0 +1,284 @@
+import * as _ from 'lodash';
+
+import {
+    ConstructorAbi,
+    DataItem,
+    DevdocOutput,
+    EventAbi,
+    FallbackAbi,
+    MethodAbi,
+    StandardContractOutput,
+} from 'ethereum-types';
+
+import { Compiler, CompilerOptions } from '@0xproject/sol-compiler';
+import {
+    DocAgnosticFormat,
+    DocSection,
+    Event,
+    EventArg,
+    Parameter,
+    SolidityMethod,
+    Type,
+    TypeDocTypes,
+} from '@0xproject/types';
+
+/**
+ * Invoke the Solidity compiler and transform its ABI and devdoc outputs into
+ * the types that are used as input to documentation generation tools.
+ * @param contractsToCompile list of contracts for which to generate doc objects
+ * @param contractsDir the directory in which to find the `contractsToCompile` as well as their dependencies.
+ * @return doc object for use with documentation generation tools.
+ */
+export async function generateSolDocAsync(
+    contractsToCompile: string[],
+    contractsDir: string,
+): Promise<DocAgnosticFormat> {
+    const doc: DocAgnosticFormat = {};
+
+    const compilerOptions = _makeCompilerOptions(contractsToCompile, contractsDir);
+    const compiler = new Compiler(compilerOptions);
+    const compilerOutputs = await compiler.getCompilerOutputsAsync();
+    for (const compilerOutput of compilerOutputs) {
+        const contractFileNames = _.keys(compilerOutput.contracts);
+        for (const contractFileName of contractFileNames) {
+            const contractNameToOutput = compilerOutput.contracts[contractFileName];
+
+            const contractNames = _.keys(contractNameToOutput);
+            for (const contractName of contractNames) {
+                const compiledContract = contractNameToOutput[contractName];
+                if (_.isUndefined(compiledContract.abi)) {
+                    throw new Error('compiled contract did not contain ABI output');
+                }
+                doc[contractName] = _genDocSection(compiledContract);
+            }
+        }
+    }
+
```

Comment:
Nit: unnecessary new line. We try to use new lines to delineate separate chunks of logic that are related.

Reply:
I use newlines like that too, and I used it in that way here.  This method has 3 newline-separated "sections", one is declaration of the variable to be returned, the next is initialization of that variable, and the last is the return of that variable.

Note that I followed this same pattern in several other methods (`_makeCompilerOptions()`, `_genDocSection()`, and `_genConstructorDoc`, to name a few), but looking back I can see that I wasn't consistent about it!